libraries/AP_Airspeed/AP_Airspeed_ASP5033.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+  backend driver for airspeed from www.qio-tek.com a I2C ASP5033 sensor
+ */
+
+#include "AP_Airspeed_ASP5033.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+extern const AP_HAL::HAL &hal;
+
+#define ASP5033_I2C_ADDR 0x6D
+#define SYS_CONFIG       0xA5
+#define CMD              0x30
+#define CMD_MEASURE      0x0A
+#define PRESS_DATA       0x06
+#define TEMP_DATA        0x09
+#define WAI_REG          0x01
+#define WAI_SET          0XA4
+#define DEVICE_ID        0x66
+#define SENSOR_READY     0x08
+
+
+AP_Airspeed_ASP5033::AP_Airspeed_ASP5033(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_ASP5033::init()
+{
+    uint8_t SysID;
+    uint8_t whoami;
+    _dev = hal.i2c_mgr->get_device(get_bus(), ASP5033_I2C_ADDR);
+    if (!_dev) {
+        return false;
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO,"ASP5033: sensor not found\n");
+    }
+    _dev->get_semaphore()->take_blocking();
+    _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+    _dev->write_register(WAI_SET, DEVICE_ID); //Set Whoami ID. 0XA4 is a OPT WhoamI Part ID register, also can be read from address 0x01.
```

Comment:
Why write the whoami? That sorta defeats the point of an identification routine as if it's not the right sensor you just altered the configuration of a different sensor. (Indeed for the FBM320 this address is actually part of the calibration data, which hopefully is a read only register).

Reply:
ASP5033 & FBM320 0x01 register default value are also 0x00, we can write a new partID to 0x40 register that change the partID to what we want, partID it can be read from address 0x01 after changed,it could be make a distinction for FMB320.

Reply:
You are writing to 0xA4, which is a calibration register on the FBM320. Not to 0x01.

Reply:
They are two different sensors, write 0x66 to 0xA4 register to FBM320, it can't chage FBM320 whomai ID to 0x66, Yep on FBM320 0XA4 is calibration register, but don't forget this register it's read only, but on the asp5033 this operation can be changed 0x01 register default value 0x00 to 0x66, so we can distinguish between two different sensors.

Reply:
They are two different sensors, write 0x66 to 0xA4 register to FBM320, it can't chage FBM320 whomai ID to 0x66, Yep on FBM320 0XA4 is calibration register, but don't forget this register it's read only, but on the asp5033 this operation can be changed 0x01 register default value 0x00 to 0x66, so we can distinguish between two different sensors.